exchanges/kucoin/kucoin_websocket.go

```diff
@@ -964,15 +970,6 @@ func (ku *Kucoin) processMarketSnapshot(respData []byte, instrument string) erro
 		Close:        response.Data.Close,
 		LastUpdated:  response.Data.Datetime.Time(),
 	}
-	assetEnabledPairs := ku.listOfAssetsCurrencyPairEnabledFor(pair)
-	if assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil {
-		ku.Websocket.DataHandler <- &spotTickerPrice
-	}
-	if assetEnabledPairs[asset.Margin] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Margin) && ku.CurrencyPairs.IsAssetEnabled(asset.Margin) == nil {
-		marginTickerPrice := spotTickerPrice
-		marginTickerPrice.AssetType = asset.Margin
-		ku.Websocket.DataHandler <- &marginTickerPrice
-	}
```

Comment:
Apologies if I was not clear in the previous comment. Please add these back. This allows the DataHandler to process the ticker for both spot and margin at the same time since they share the same data feed. Since you've now added in variables, it can look like the following:

```go
        if assetEnabledPairsSpot {
		ku.Websocket.DataHandler <- &spotTickerPrice
	}
	if assetEnabledPairsMargin {
		marginTickerPrice := spotTickerPrice
		marginTickerPrice.AssetType = asset.Margin
		ku.Websocket.DataHandler <- &marginTickerPrice
	}
```

Reply:
Completely misunderstood, sorry about that. Added an if else statement in case the asset is both Margin and Spot

Reply:
Please just make it another `if` on its own like my above example. By being an `else if`, it cannot populate both spot and margin

Reply:
amended, I have changed the test slightly so it covers different cases of the assets